pkg/parse/parse.go

```diff
@@ -48,6 +49,22 @@ const (
 	BuildahCacheDir = "buildah-cache"
 )
 
+// RepoNameToNamedReference parse the raw string to Named reference
+func RepoNameToNamedReference(dest string) (reference.Named, error) {
+	ref, err := reference.Parse(dest)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse reference: %w", err)
+	}
+	named, ok := ref.(reference.Named)
+	if !ok {
+		return nil, fmt.Errorf("invalid repo %q: must contain registry and repository", dest)
+	}
+	if !reference.IsNameOnly(named) {
+		return nil, fmt.Errorf("invalid parsed named: %v", named)
```

Comment:
This is an error case users might actually reasonably trigger, and the message needs to be useful to users (e.g. “repository must contain neither a tag nor digest”).